merkle/src/tree_impl.rs

```diff
@@ -78,6 +100,65 @@ where
         }
     }
 
+    /// Gets the merkle proof of an item with the 0-based index of the item
+    /// being added, e.g. the first item is index 0.
+    /// This function panics when the index is out of range.
+    pub fn get_proof(
+        &mut self,
+        index: usize,
+    ) -> Option<DefaultMerkleProof<Item, Hash, Hasher, Merger>> {
```

Comment:
This function body is really hard to read and understand for me. I think it could benefit from some comments and possibly rearranging a bit for readibility

Reply:
Added comments to describe the flow, plz let me know whether it helps